### PR TITLE
Settlement module with test case

### PR DESCRIPTION
**Pull request by sonyy-m**
Ref: [https://github.com/razorpay/razorpay-php/blob/master/documents/settlement.md](https://github.com/razorpay/razorpay-php/blob/master/documents/settlement.md)
Api Doc: [https://razorpay.com/docs/api/settlements/](https://razorpay.com/docs/api/settlements/)

### CODE REVIEW

File: constants/url.go [hunk 1]
```diff
@@ -27,4 +27,6 @@ const (
 	SUBSCRIPTION_URL = "/subscriptions"
 	//PLAN_URL ... plan api base path
 	PLAN_URL = "/plans"
+	//SETTLEMENT_URL ... settlements api base path
+    SETTLEMENT_URL = "/settlements"
```
Comment (Madeyedexter) [inline constants/url.go:31]: Fix indentation

File: resources/settlement.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+// All fetches collection of settlements for the given queryParams.
+func (settlement *Settlement) All(queryParams map[string]interface{}, extraHeaders map[string]string) (map[string]interface{}, error) {
+	return settlement.Request.Get(constants.SETTLEMENT_URL, queryParams, extraHeaders)
+}
+
+// Fetch fetches a settlement having the given settlementID.
+func (settlement *Settlement) Fetch(settlementID string, queryParams map[string]interface{}, extraHeaders map[string]string) (map[string]interface{}, error) {
+	url := fmt.Sprintf("%s/%s", constants.SETTLEMENT_URL, settlementID)
```
Comment (Madeyedexter) [inline resources/settlement.go:22]: Please url encode settlement id

**Review by Madeyedexter — COMMENTED** [review]
Please ensure EOL character at the end of each json file

File: constants/url.go (hunk 1 above)
Comment (sonyy-m) [reply constants/url.go]: done

Comment (sonyy-m) [reply constants/url.go]: done

**Review by Madeyedexter — APPROVED** [review]

**Pull request merged**